rust/lance-index/src/scalar.rs

```diff
@@ -554,6 +555,43 @@ impl AnyQuery for TextQuery {
     }
 }
 
+/// A query that a InvertedIndex can satisfy
+#[derive(Debug, Clone, PartialEq)]
+pub enum TokenQuery {
+    /// Retrieve all row ids where the text contains all tokens parsed from given string. The tokens
```

Comment:
There might be a few pieces of glue missing.  I think we need a `TokenQueryParser` (see https://github.com/lancedb/lance/blob/60711f360b7f8692df44a0e84c98c8fdff2897a3/rust/lance-index/src/scalar/expression.rs#L348 for the `TextQueryParser`).

We also need to register the token query parser here: https://github.com/lancedb/lance/blob/60711f360b7f8692df44a0e84c98c8fdff2897a3/rust/lance/src/index.rs#L1361

Can we call `is_query_allowed` in the registration function (`scalar_index_info`)?  This way we can skip the scalar index entirely if it is not eligible.  Returning `AtLeast` with zero rows might lead to bad performance (the planner will think we are doing a scalar index optimized search and make certain decisions based on that)

Reply:
> I think we need a TokenQueryParser

We  already have a FtsQueryParser which parses ```contains_tokens``` into TokenQuery::TokensContains. Actually you implemented it (^-^). I think we can just rely on FtsQueryParser.

> Can we call is_query_allowed in the registration function ...

Thanks your nice suggestion, let me fix it.